cpp/src/arrow/util/rle_encoding_test.cc

```diff
@@ -173,6 +173,39 @@ TEST(BitArray, TestMixed) {
   }
 }
 
+// Writes 'num_vals' values with width 'bit_width' and reads them back.
+static void TestPutValue(int bit_width, uint64_t num_vals) {
+  const uint64_t max = uint64_t(2) << (bit_width - 1);
+  num_vals = std::min(num_vals, max);
+  int len = static_cast<int>(bit_util::BytesForBits(bit_width * num_vals));
+  EXPECT_GT(len, 0);
+
+  std::vector<uint8_t> buffer(len);
+  bit_util::BitWriter writer(buffer.data(), len);
+  for (uint64_t i = max - num_vals; i < max; i++) {
+    bool result = writer.PutValue(i, bit_width);
+    EXPECT_TRUE(result);
+  }
+  writer.Flush();
+  EXPECT_EQ(writer.bytes_written(), len);
+
+  bit_util::BitReader reader(buffer.data(), len);
+  for (uint64_t i = max - num_vals; i < max; i++) {
+    int64_t val = 0;
+    bool result = reader.GetValue(bit_width, &val);
+    EXPECT_TRUE(result);
+    EXPECT_EQ(val, i);
+  }
+  EXPECT_EQ(reader.bytes_left(), 0);
+}
+
+TEST(BitUtil, RoundTripIntValues) {
+  for (int width = 1; width < 64; width++) {
```

Comment:
Ideally this should be:
```suggestion
  for (int width = 1; width <= 64; width++) {
```

Reply:
Reverted this because it causes an overflow to 0 and testing for `bit_width == 0` doesn't make sense I think.

Reply:
What do you mean? Testing that we are able to write 64-bit values makes sense.

Perhaps you want something like:
```c++
  // The max value representable in `bit_width` bits.
  const uint64_t max = std::numeric_limits<uint64_t>::max() >> (64 - bit_width);
```

Reply:
Changed to your proposal. It runs ok locally.

Reply:
> What do you mean? Testing that we are able to write 64-bit values makes sense.

What I meant was that `2 << 63` was overflowing, setting `max` and hence `num_vals` to `0`. That then tripped the `EXPECT_GT(len, 0);` test. This is not an issue now I expect.